api/bootstrap.proto

```diff
@@ -17,91 +17,122 @@ import "google/protobuf/duration.proto";
-    // trace drivers.
-    string name = 1;
+    // supported HTTP trace driver. *envoy.lightstep* and *envoy.zipkin* are
+    // built-in trace drivers.
+    string name = 1 [(validate.rules).string.min_len = 1];
     // Trace driver specific configuration which depends on the driver being
-    // instantiated. See the trace drivers for further documentation.
+    // instantiated. See the :ref:`LightstepConfig
+    // <envoy_api_msg_LightstepConfig>` and :ref:`ZipkinConfig
+    // <envoy_api_msg_ZipkinConfig>` trace drivers for examples.
     google.protobuf.Struct config = 2;
   }
+  // Provides configuration for the HTTP tracer.
   Http http = 1;
 }
 
+// Administration interface :ref:`operations documentation
+// <operations_admin_interface>`.
 message Admin {
   // The path to write the access log for the administration server. If no
   // access log is desired specify ‘/dev/null’.
-  string access_log_path = 1;
+  string access_log_path = 1 [(validate.rules).string.min_len = 1];
+
   // The cpu profiler output path for the administration server. If no profile
   // path is specified, the default is ‘/var/log/envoy/envoy.prof’.
   string profile_path = 2;
+
   // The TCP address that the administration server will listen on.
-  Address address = 3;
+  Address address = 3 [(validate.rules).message.required = true];
 }
 
+// Cluster manager :ref:`architecture overview <arch_overview_cluster_manager>`.
 message ClusterManager {
   // Name of the local cluster (i.e., the cluster that owns the Envoy running
-  // this configuration). In order to enable zone aware routing this option must
-  // be set. If local_cluster_name is defined then clusters must contain a
-  // definition of a cluster with the same name.
+  // this configuration). In order to enable :ref:`zone aware routing
+  // <arch_overview_load_balancing_zone_aware_routing>` this option must be set.
+  // If *local_cluster_name* is defined then :ref:`clusters
+  // <config_cluster_manager_clusters>` must be defined in the :ref:`Bootstrap
+  // static cluster resources
+  // <envoy_api_field_Bootstrap.StaticResources.clusters>`.
   string local_cluster_name = 1;
 
-  // Optional global configuration for outlier detection.
   message OutlierDetection {
+    // Specifies the path to the outlier event log.
     string event_log_path = 1;
   }
+  // Optional global configuration for outlier detection.
   OutlierDetection outlier_detection = 2;
 
   // Optional configuration used to bind newly established upstream connections.
   // This may be overridden on a per-cluster basis by upstream_bind_config in the cds_config.
   BindConfig upstream_bind_config = 3;
 
-  // A management server endpoint to stream load stats to, as per
-  // StreamLoadStats in eds.proto. This must have api_type GRPC.
+  // A management server endpoint to stream load stats to via
+  // :ref:`StreamLoadStats <envoy_api_msg_LoadStatsRequest>`.  This must have
```

Comment:
two space after period (we should lint for this everywhere in the code base). ;)

Reply:
I believe @alyssawilk is working on a script for this :+1:

Reply:
Alas https://github.com/envoyproxy/envoy/pull/2130 only works on the Envoy side of things.  I don't think we run any sort of clean up on the data plane API.   
If we think it's worth our time I'll add check_format scripts over here if someone else will add the presubmit so we all don't forget it constantly... :-)

Reply:
There is already an issue in this repo for check/fix format. We need it for buildifier also, so doing this would be awesome.